pkg/splunk/enterprise/indexercluster.go

```diff
@@ -184,8 +186,11 @@ func ApplyIndexerClusterManager(ctx context.Context, client splcommon.Controller
 	for _, v := range statefulsetPods.Items {
 		for _, owner := range v.GetOwnerReferences() {
 			if owner.UID == statefulSet.UID {
+				previousImage := v.Spec.Containers[0].Image
+				currentImage := cr.Spec.Image
 				// get the pod image name
-				if v.Spec.Containers[0].Image != cr.Spec.Image {
+				if strings.HasPrefix(previousImage, "8") &&
+					strings.HasPrefix(currentImage, "9") {
```

Comment:
Can we avoid constant values here? 
Nit: Probably, adding a function call may be more readable?

Reply:
this code been there from few cycles , yes I will create a function just to make sure its one time activity